internal/pkg/ai-functions/handler/handler_ai_function.go

```diff
@@ -93,50 +166,56 @@ func (h *AIFunction) Apply(ctx context.Context, req *pb.ApplyRequest) (pbValue *
 		r.Header.Set("X-Ai-Proxy-Org-Id", base64.StdEncoding.EncodeToString([]byte(apis.GetOrgID(ctx))))
 		r.Header.Set("X-Ai-Proxy-User-Id", base64.StdEncoding.EncodeToString([]byte(apis.GetUserID(ctx))))
 	})
-	client, err := sdk.NewClient(h.OpenaiURL, http.DefaultClient, ros...)
+	client, err := sdk.NewClient(openaiURL, http.DefaultClient, ros...)
 	if err != nil {
-		return nil, err
+		return err
 	}
 	completion, err := client.CreateCompletion(ctx, options)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to CreateCompletion")
+		return errors.Wrap(err, "failed to CreateCompletion")
 	}
 	if len(completion.Choices) == 0 || completion.Choices[0].Message == nil || completion.Choices[0].Message.FunctionCall == nil {
-		return nil, errors.New("no idea") // todo: do not return error, response friendly
+		return errors.New("no idea") // todo: do not return error, response friendly
 	}
 	// todo: check index out of range and invalid memory reference
 	arguments := completion.Choices[0].Message.FunctionCall.JSONMessageArguments()
 	if err = fd.VerifyArguments(arguments); err != nil {
-		return nil, errors.Wrap(err, "invalid arguments from FunctionCall")
+		return errors.Wrap(err, "invalid arguments from FunctionCall")
+	}
+
+	input := aitestcase.TestCaseFunctionInput{
+		TestSetID: req.GetFunctionParams().TestSetID,
+		IssueID:   tp.IssueID,
+		Prompt:    tp.GetPrompt(),
 	}
-	result, err := f.Callback(ctx, arguments)
+	result, err := f.Callback(ctx, arguments, input, req.NeedAdjust)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to Callback with arguments: %s", string(arguments))
+		return errors.Wrapf(err, "failed to Callback with arguments: %s", string(arguments))
 	}
-	var v = &structpb.Value{}
-	switch i := result.(type) {
-	case string:
-		if err := jsonpb.UnmarshalString(i, v); err != nil {
-			return nil, err
-		}
-	case []byte:
-		if err := jsonpb.UnmarshalString(string(i), v); err != nil {
-			return nil, err
-		}
-	case json.RawMessage:
-		if err := jsonpb.UnmarshalString(string(i), v); err != nil {
-			return nil, err
-		}
-	default:
-		var buf bytes.Buffer
-		if err := json.NewEncoder(&buf).Encode(result); err != nil {
-			return nil, err
+	*results = append(*results, result)
+	return nil
+}
+
+// validateParamsForCreateTestcase 校验创建测试用例对应的参数配置
+func validateParamsForCreateTestcase(req *pb.ApplyRequest) error {
```

Comment:
validate at proto side.

Reply:
use  ’google.protobuf.Value functionParams‘ now， so can not validate at proto side